odoo/models.py

```diff
@@ -2046,6 +2040,15 @@ def _read_group_raw(self, domain, fields, groupby, offset=0, limit=None, orderby
             )
         return result
 
+    def _read_group_resolve_many2one_fields(self, data, fields):
+        many2onefields = [field['field'] for field in fields if field['type'] == 'many2one']
```

Comment:
Could probably be a genexp rather than a listcomp, after your refactoring it's only iterated once so the list is unnecessary.